GPSUARTDriver.device.lib.nut

```diff
@@ -0,0 +1,110 @@
+// MIT License
+// Copyright 2017-18 Electric Imp
+// SPDX-License-Identifier: MIT
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+// Driver for UART GPS modules. This driver is dependant on the GPSParser class.
+// This driver is focused on retrieving location data and returns data from the
+// GPSParser.getGPSDataTable method. Currently only the data with sentence IDs
+// VTG, RMC, GLL, GGA, GSV and GSA are supported.
+class GPSUARTDriver {
+
+    function _statics_() {
+        const VERSION          = "1.0.0";
+        const LINE_MAX         = 150;
+        const DEFAULT_BAUDRATE = 9600;
+    }
+
+    _gps            = null;
+    _gpsLine        = null;
+    _callback       = null;
+
+    _hasFix         = null;
+    _lastLat        = null;
+    _lastLong       = null;
+
+    constructor(uart, baudrate = null, dataReady = null) {
+        _gps     = uart;
+        _hasFix  = false;
+        _gpsLine = "";
+
+        // Check optional params
```

Comment:
to be honest I won't do so. If you don't want to specify boudrate, just pass null as the second parameter. It's much easier to explain/understand and is not too much overhead on users overall. Do you agree?

Reply:
I'll make it a required parameter then. I'll use the default if only if someone passes in a non-integer